From: Raymond Wooninck <tittiatcoke@gmail.com>
Date: Sat, 04 Oct 2014 18:15:21 +0000
Subject: Register the metatypes for the plasmoid as that those were removed from PackageKit-qt itself
X-Git-Url: http://quickgit.kde.org/?p=apper.git&a=commitdiff&h=75b3d7d1587d4fe66ea8d157efc022f20d853dc1
---
Register the metatypes for the plasmoid as that those were removed from PackageKit-qt itself

CCMAIL: dantti12@gmail.com
---


--- a/declarative-plugins/qmlplugins.cpp
+++ b/declarative-plugins/qmlplugins.cpp
@@ -50,6 +50,25 @@
     qmlRegisterType<PackageKit::Transaction>(uri, 0, 1, "Transaction");
     qmlRegisterUncreatableType<PackageKit::Daemon>(uri, 0, 1, "Daemon", "Global");
     qRegisterMetaType<PkTransaction::ExitStatus>("PkTransaction::ExitStatus");
+    qRegisterMetaType<PackageKit::Daemon::Network>("PackageKit::Daemon::Network");
+    qRegisterMetaType<PackageKit::Daemon::Authorize>("PackageKit::Daemon::Authorize");
+    qRegisterMetaType<PackageKit::Transaction::InternalError>("PackageKit::Transaction::InternalError");
+    qRegisterMetaType<PackageKit::Transaction::Role>("PackageKit::Transaction::Role");
+    qRegisterMetaType<PackageKit::Transaction::Error>("PackageKit::Transaction::Error");
+    qRegisterMetaType<PackageKit::Transaction::Exit>("PackageKit::Transaction::Exit");
+    qRegisterMetaType<PackageKit::Transaction::Filter>("PackageKit::Transaction::Filter");
+    qRegisterMetaType<PackageKit::Transaction::Message>("PackageKit::Transaction::Message");
+    qRegisterMetaType<PackageKit::Transaction::Status>("PackageKit::Transaction::Status");
+    qRegisterMetaType<PackageKit::Transaction::MediaType>("PackageKit::Transaction::MediaType");
+    qRegisterMetaType<PackageKit::Transaction::DistroUpgrade>("PackageKit::Transaction::DistroUpgrade");
+    qRegisterMetaType<PackageKit::Transaction::TransactionFlag>("PackageKit::Transaction::TransactionFlag");
+    qRegisterMetaType<PackageKit::Transaction::TransactionFlags>("PackageKit::Transaction::TransactionFlags");
+    qRegisterMetaType<PackageKit::Transaction::Restart>("PackageKit::Transaction::Restart");
+    qRegisterMetaType<PackageKit::Transaction::UpdateState>("PackageKit::Transaction::UpdateState");
+    qRegisterMetaType<PackageKit::Transaction::Group>("PackageKit::Transaction::Group");
+    qRegisterMetaType<PackageKit::Transaction::Info>("PackageKit::Transaction::Info");
+    qRegisterMetaType<PackageKit::Transaction::SigType>("PackageKit::Transaction::SigType");
+
 }
 
 void QmlPlugins::initializeEngine(QDeclarativeEngine *engine, const char *uri)

